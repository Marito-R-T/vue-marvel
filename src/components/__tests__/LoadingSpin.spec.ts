import { describe, it, expect } from 'vitest'

import { mount } from '@vue/test-utils';
import LoadingSpin from '../LoadingSpin.vue'; // Make sure the component path is correct

describe('LoadingSpin.vue', () => {
  it('renders the loading spinner', () => {
    const wrapper = mount(LoadingSpin);

    // Check if the component has been rendered successfully
    expect(wrapper.exists()).toBe(true);

    // Check if the component contains the "loading-container" class
    expect(wrapper.find('.loading-container').exists()).toBe(true);

    // Check if the component contains the "loading-spinner" class
    expect(wrapper.find('.loading-spinner').exists()).toBe(true);
  });

  it('matches snapshot', () => {
    const wrapper = mount(LoadingSpin);

    // Compare the component's snapshot to a previously saved version
    expect(wrapper.html()).toMatchSnapshot();
  });
});
