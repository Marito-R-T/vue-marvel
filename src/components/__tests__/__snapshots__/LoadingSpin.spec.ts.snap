// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`LoadingSpinner.vue > matches snapshot 1`] = `
"<div class=\\"loading-container\\">
  <div class=\\"loading-spinner\\"></div>
</div>"
`;
